### ISSUE TITLE
Manually give initial estimates of calibration parameters.

### ISSUE THREAD

**Issue by johnny04thomas** [ethz-asl/kalibr#367]
I would like to manually input initial estimates of calibration parameters (intrinsic and distortion) from previously calibrated values in Multiple camera calibration. Please help me identify the file and line to modify for fisheye models: omni-radtan and ds. Thanking in advance.

**Comment by haoyuanz13** [comment]
any update here?

**Comment by johnny04thomas** [comment]
Yes, I managed to do that. Sorry for not updating about it before. In kalibr/aslam_offline_calibration/kalibr/python/kalibr_calibrate_cameras once the initial individual intrinsic parameters are evaluated, you could set your calibration parameters as the initial guesses for the intrinsics inside the main while loop. 
    ```
    cameraList[0].geometry.projection().setParameters(np.array(projection_parameters))
    cameraList[0].geometry.projection().distortion().setParameters(np.array(distortion_parameters))
```
After that the intrinsics initialized are used as known parameters in the optimization by deactivating their role as the design variable. The following modification is required to be made inside **CameraCalibrator.py** file in **kalibr_camera_calibration** folder.
    `177   camera.setDvActiveStatus(False, False, False)`
I hope this helps.

**Comment by haoyuanz13** [comment]
Good to KNOW that!

I have tried your instructions and it works which fixes the initial intrinsic params during the optimization, thanks for your reply!

**Comment by johnny04thomas** [comment]
You're Welcome!

**Comment by ishipachev** [comment]
@johnny04thomas 
Thank you. It helped me too.

**Comment by foemre** [comment]
Could be worthwhile to consider adding the option for manual entry of initial guesses.
